test_vm/tests/commit_post_test.rs

```diff
@@ -222,35 +239,16 @@ fn skip_sector() {
     );
 
     // miner still has initial pledge
-    let balances = v.get_miner_balance(&miner_info.miner_id);
+    let balances = miner_balance(v, &miner_info.miner_id);
     assert!(balances.initial_pledge.is_positive());
-
-    // power unproven so network stats are the same
-    let network_stats = v.get_network_stats();
-    assert!(network_stats.total_bytes_committed.is_zero());
-    assert!(network_stats.total_pledge_collateral.is_positive());
-
-    v.assert_state_invariants();
 }
 
 #[test]
-fn missed_first_post_deadline() {
+fn missed_first_post_deadline_testvm() {
     let store = MemoryBlockstore::new();
-    let (mut v, miner_info, sector_info) = setup(&store);
+    let (v, miner_info, sector_info) = setup(&store);
 
-    // move to proving period end
-    v = v.with_epoch(sector_info.deadline_info.last());
-
-    // Run cron to detect missing PoSt
-
-    apply_ok(
-        &v,
-        &SYSTEM_ACTOR_ADDR,
-        &CRON_ACTOR_ADDR,
-        &TokenAmount::zero(),
-        CronMethod::EpochTick as u64,
-        None::<RawBytes>,
-    );
+    missed_first_post_deadline(&v, sector_info);
```

Comment:
The inner test can run with an injected VM but ExpectInvocation must run on the concrete `TestVM`